base_multi_search_separator/monkeypatching.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 Akretion (http://www.akretion.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import re
+
+from odoo.models import BaseModel
+
+search_original = BaseModel.search
+
+
+def search(self, domain, offset=0, limit=None, order=None, count=False):
+    """Override of the Python method to remove the dependency of the unit
+    fields"""
+    if self != self.env["res.lang"]:
+        list_separator = self.env["ir.config_parameter"].get_param(
+            "multi_search_separator_base"
+        )
+    else:
+        list_separator = False
+    if not list_separator or not domain:
+        return search_original(self, domain, offset, limit, order, count)
+
+    domain_list = []
+    domain_value = []
+    nbr = 0
+    nlist_sep = []
+    for sep in list_separator:
+        nlist_sep.append(sep)
+    for dom in domain:
+        if (
+            len(dom) == 3
+            and isinstance(dom[2], str)
+            and any(separator in dom[2] for separator in list_separator)
+        ):
+            value_list = re.split("{}".format(nlist_sep), dom[2])
+            value_list.append(dom[2])
+            value_list_set = set(value_list)
+            for separator in list_separator:
+                value_list_set = value_list_set.union(dom[2].split(separator))
+            for value in value_list_set:
+                domain_search_field = (dom[0], dom[1], value)
+                nbr = nbr + 1
+                domain_value.append(domain_search_field)
+            nbr = nbr - 1
+        else:
+            domain_list.append(dom)
+    for _ in range(nbr):
+        domain_list.append("|")
```

Comment:
I think you can manipulate all the domain with expression.OR and expression.AND. It will be more readable I think.